mllib/src/test/scala/org/apache/spark/ml/regression/DecisionTreeRegressorSuite.scala

```diff
@@ -57,6 +58,29 @@ class DecisionTreeRegressorSuite
     compareAPIs(categoricalDataPointsRDD, dt, categoricalFeatures)
   }
 
+  test("SPARK-18036: Verify Decision Tree regression throws" +
```

Comment:
Take a look at [this test](https://github.com/apache/spark/blob/master/mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala#L164). It should have failed before, but it didn't due to how we handle different types of features differently.

I prefer not to add tests to each of DecisionTree, GBT, RF suites, and instead add a single test to `RandomForestSuite` which exercises constant features with both exclusively continuous features and a mixture of ordered categorical, unordered categorical, and continuous features.